src/andromede/study/resolve_components.py

```diff
@@ -0,0 +1,165 @@
+    port_ref_1 = PortRef(component_1, port1)
+    port_ref_2 = PortRef(component_2, port2)
+
+    return PortsConnection(port_ref_1, port_ref_2)
+
+
+def _get_component_by_id(
+    all_components: List[Component], component_id: str
+) -> Optional[Component]:
+    components_dict = {component.id: component for component in all_components}
+    return components_dict.get(component_id)
+
+
+def consistency_check(
+    input_components: Dict[str, Component], input_models: Dict[str, Model]
+) -> bool:
+    """
+    Checks if all components in the Components instances have a valid model from the library.
+    Returns True if all components are consistent, raises ValueError otherwise.
+    """
+    model_ids_set = input_models.keys()
+    for component_id, component in input_components.items():
+        if component.model.id not in model_ids_set:
+            raise ValueError(
+                f"Error: Component {component_id} has invalid model ID: {component.model.id}"
+            )
+    return True
+
+
+def build_network(comp_network: NetworkComponents) -> Network:
+    network = Network("study")
+
+    for node_id, node in comp_network.nodes.items():
+        node = Node(model=node.model, id=node_id)
+        network.add_node(node)
+
+    for component_id, component in comp_network.components.items():
+        network.add_component(component)
+
+    for connection in comp_network.connections:
+        network.connect(connection.port1, connection.port2)
+    return network
+
+
+def build_data_base(input_comp: InputComponents) -> DataBase:
+    database = DataBase()
+    for comp in input_comp.components:
+        for param in comp.parameters or []:
+            param_value = _evaluate_param_type(
+                param.type, param.value, param.timeseries
+            )
+            database.add_data(comp.id, param.name, param_value)
+
+    return database
+
+
+def _evaluate_param_type(
+    param_type: str, param_value: Optional[float], timeseries: Optional[str]
+) -> AbstractDataStructure:
+    if param_type == "constant" and param_value is not None:
+        return ConstantData(float(param_value))
+
+    elif param_type == "timeseries":
+        return TimeScenarioSeriesData(_evaluate_time_series(timeseries))
+
+    raise ValueError(f"Data should be either constant or timeseries ")
+
+
+def _evaluate_time_series(file_ts: Optional[str]) -> Dict[TimeScenarioIndex, float]:
+    """Read time series .txt file delimited by tab"""
+    time_series = {}
+    try:
+        if file_ts is not None:
+            path = Path(file_ts)
+            df = pd.read_csv(path, header=None)
+            values = df.values
+            arr_split = [x[0].split() for x in values]
+            num_rows = len(arr_split)
+            num_cols = len(arr_split[0])
+            for time in range(num_rows):
+                for scenario in range(num_cols):
+                    index = TimeScenarioIndex(time=time, scenario=scenario)
```

Comment:
Actually the best thing would be to change `TimeScenarioSeriesData` to just put the pandas table into it.

Reply:
Yes easier to have directly the dataFrame